statsmodels/regression/linear_model.py

```diff
@@ -773,6 +668,189 @@ def whiten(self, Y):
         """
         return Y
 
+
+    def score(self, params, scale=None):
+        """
+        Evaluate the score function at a given point.
+
+        The score corresponds to the profile (concentrated)
+        log-likelihood in which the scale parameter has been profiled
+        out.
+
+        Parameters
+        ----------
+        params : array-like
+            The parameter vector at which the score function is
+            computed.
+        scale : float or None
+            If None, return the profile (concentrated) log likelihood
+            (profiled over the scale parameter), else return the
+            log-likelihood using the given scale value.
+
+        Returns
+        -------
+        The score vector.
+        """
+
+        if not hasattr(self, "_wexog_xprod"):
+            self._setup_score_hess()
+
+        xtxb = np.dot(self._wexog_xprod, params)
+        sdr = -self._wexog_x_wendog + xtxb
+
+        if scale is None:
+            ssr = self._wendog_xprod - 2 * np.dot(self._wexog_x_wendog.T, params)
+            ssr += np.dot(params, xtxb)
+            return -self.nobs * sdr / ssr
+        else:
+            return -sdr / scale
+
+
+    def _setup_score_hess(self):
+        y = self.wendog
+        if hasattr(self, 'offset'):
+            y = y - self.offset
+        self._wendog_xprod = np.sum(y * y)
+        self._wexog_xprod = np.dot(self.wexog.T, self.wexog)
+        self._wexog_x_wendog = np.dot(self.wexog.T, y)
```

Comment:
in terms of names, I like the `t` for transpose in the middle  xtx, xty (for internal)
`_wexog_t_wendog` ?